sunbeam/jobs/juju.py

```diff
@@ -270,6 +290,104 @@ async def remove_unit(self, name: str, unit: str, model: str):
 
         await application.destroy_unit(unit)
 
+    @controller
+    async def get_leader_unit(self, name: str, model: str) -> str:
+        """Get leader unit.
+
+        :name: Application name
+        :model: Name of the model where the application is located
+        :returns: Unit name
+        """
+        model_impl = await self.get_model(model)
+
+        try:
+            for unit in model_impl.applications[name].units:
+                is_leader = await unit.is_leader_from_status()
+                if is_leader:
+                    return unit.entity_id
+        except KeyError:
+            raise ApplicationNotFoundException(
+                f"Application {name!r} is missing from model {model!r}"
+            )
```

Comment:
This can be simplified to the following, since get_application will take care of raising the not found exception

```suggestion
        application = await self.get_application(name, model)

        for unit in application.units:
            is_leader = await unit.is_leader_from_status()
            if is_leader:
                return unit.entity_id

        raise LeaderNotFoundException(
            f"Leader for application {name!r} is missing from model {model!r}"
        )
```

Reply:
Fixed in latest push